omegatrade-app/frontend/src/app/components/stock-dashboard/stock-dashboard.component.ts

```diff
@@ -0,0 +1,182 @@
+import { RestService } from '../../services/rest.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { interval, Subscription } from "rxjs";
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-stock-dashboard',
+  templateUrl: './stock-dashboard.component.html',
+  styleUrls: ['./stock-dashboard.component.css']
+})
+export class StockDashboardComponent implements OnInit {
+
+  selectedCompany: string = "";
+  lastUpdatedTime;
+  stock: StockChart;
+  stocks: any;
+  company: any;
+  companies: any;
+  loader: boolean = false;
+  subscription: Subscription;
+  intervalId: number;
+  sId: String;
+
+  constructor(private router: ActivatedRoute, private _snackBar: MatSnackBar, private restService: RestService) {
+  }
+
+  ngOnInit(): void {
+    this.sId = this.router.snapshot.queryParamMap.get('sId');
+    this.getCompanies();
+  }
+
+  getCompanies() {
+    this.restService.getData('companies/list')
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.companies = response.data;
+            if (this.companies && this.companies.length > 0) {
+              if(this.sId){
+                this.selectedCompany = this.router.snapshot.queryParamMap.get('companyId');
+              }else{
+                this.selectedCompany = this.companies[0].companyId;
+              }
+              this.getStockData();
+            }
+          }
+        },
+        error => {
+          this.openSnackBar(error.error.message, "")
+          this.loader = false;
+        });
+  }
+
+  getStockData() {
+    if (this.selectedCompany) {
+      let params = `${this.selectedCompany}`;
+      if(this.sId){
+        params += `?sId=${this.sId}` ;
+      }
+      console.log(params);
+      this.loader = true;
+      this.restService.getData(`companies/dashboard/${params}`)
+        .subscribe(
+          response => {
+            if (response && response.success) {
+              this.stocks = response.data.stocks;
+              this.company = response.data.company;
+              if (this.stocks && this.stocks.length > 0) {
+                this.parseStockDatas();
+                this.lastUpdatedTime = this.stocks[(this.stocks.length - 1)].date;
+              }else if(this.company && this.company.status === 'PROCESSING'){
+                setTimeout(() => {
+                  this.getStockData();
+              }, 5000);
+              }
+              this.loader = false;
+            }
+          },
+          error => {
+            this.openSnackBar(error.error.message, "")
+            this.loader = false;
+          });
+    }
+  }
+
+  parseStockDatas() {
+    const data = [];
+    for (var i = 0; i < this.stocks.length; i++) {
+      data.push([this.stocks[i].date, parseInt(this.stocks[i].currentValue)])
+    }
+    this.createChart(data)
+  }
+
+  createChart(data) {
+    this.stock = new StockChart({
```

Comment:
Instead of using high chart in JS way we can try to use it in angular way and similar way we can update data on chart in angular way